modules/data.atmosphere/R/download.NOAA.R

```diff
@@ -0,0 +1,387 @@
+  # Not all NOAA data units match the cf data standard.  In this next section, data are processed to
+  # confirm with the standard when necessary.
+  # The following is a list of variables which need to be processed:
+  # 1. NOAA's relative humidity must be converted to specific humidity
+  # 2. NOAA's measure of precipitation is the accumulation over 6 hours; cf's standard is precipitation per second
+  
+  if (verbose) {
+    PEcAn.logger::logger.info("Converting units.")
+  }
+  
+  #Convert NOAA's relative humidity to specific humidity
+  humid_index = cf_var_names == "specific_humidity"
+  
+  humid_data = noaa_data[humid_index][[1]]
+  temperature_data = noaa_data[cf_var_names == "air_temperature"][[1]]
+  pressure_data = noaa_data[cf_var_names == "air_pressure"][[1]]
+   
+  #Depending on the volume and dimensions of data you download, sometimes R stores it as a vector and sometimes
+  #as a matrix; the different cases must be processed with different loops.
+  if (as.logical(nrow(humid_data))) {
+     for (i in 1:length(humid_data)) {
+       humid_data[i] = PEcAn.data.atmosphere::rh2qair(humid_data[i], temperature_data[i], pressure_data[i])
+     }
+  } else {
+     for (i in 1:nrow(humid_data)) {
+       for (j in 1:ncol(humid_data)) {
+          humid_data[i,j] = PEcAn.data.atmosphere::rh2qair(humid_data[i,j], temperature_data[i,j], pressure_data[i,j])
+        }
+      }
+  }
+  
+  #Update the noaa_data list with the correct data
+  noaa_data[humid_index][[1]] <- humid_data
+  
+  # Convert NOAA's total precipitation (kg m-2) to precipitation flux (kg m-2 s-1)
+  #NOAA precipitation data is an accumulation over 6 hours.
+  precip_index = cf_var_names == "precipitation_flux"
+  
+  noaa_data[precip_index][[1]] = noaa_data[precip_index][[1]] / 21600  #There are 21600 seconds in 6 hours
+  
+  #############################################
+  # Done with data processing.  Now writing the data to the specified directory.
+  if (verbose) {
+    PEcAn.logger::logger.info("Writing data to netcdf file")
+  }
+  
+  
+  if (!dir.exists(outfolder)) {
+    dir.create(outfolder, recursive=TRUE)
+  }
+  
+  setwd(outfolder)
+  
+  #Generate file name.  File names need to account for the specificity in the data.
+  st = lubridate::year(start_date)
+  ed = lubridate::year(end_date)
+  yrcomponent = st
+  if (st != ed) { yrcomponent = paste0(st, "-", ed) } #This could be a function, but it's rather simple and specific for that.
+  
+  st = lubridate::month(start_date)
+  ed = lubridate::month(end_date)
+  moncomponent = st
+  if (st != ed) { moncomponent = paste0(st, "-", ed) }
+  
+  st = lubridate::day(start_date)
+  ed = lubridate::day(end_date)
+  dycomponent = st
+  if (st != ed) { dycomponent = paste0(st, "-", ed) }
+  
+  flname <- if (include_forecast) {
+    paste("NOAA", "GEFS", yrcomponent, moncomponent, dycomponent, "f", "nc", sep=".")
+  } else {
+    paste("NOAA", "GEFS", yrcomponent, moncomponent, dycomponent, "nc", sep=".")
+  }
```

Comment:
Per earlier comment about ISO dates, this whole bit of code can be simplifed to one case, and you don't need to hand-parse year, month, and day as `format` can do that for you. Also, seems like we probably need to include time as well not just date.

That said, right now it looks like you're setting up to write one file, rather than one file per ensemble member -- I think we need to include the ensemble member in the file name